File: solution/java/day19/src/test/java/blog/ArticleBuilder.java
package blog;

import io.vavr.collection.HashMap;
import io.vavr.collection.Map;

import static org.instancio.Instancio.create;

public class ArticleBuilder {
    public static final String AUTHOR = "Pablo Escobar";
    public static final String COMMENT_TEXT = "Amazing article !!!";
    private Map<String, String> comments;

    public ArticleBuilder() {
        comments = HashMap.empty();
    }

    public static ArticleBuilder anArticle() {
        return new ArticleBuilder();
    }

    public ArticleBuilder commented() {
        this.comments = comments.put(COMMENT_TEXT, AUTHOR);
        return this;
    }

    public Article build() {
        var article = new Article(
                create(String.class),
                create(String.class)
        );
        return comments.foldLeft(article, (a, c) -> a.addComment(c._1, c._2).get());
    }
}

File: solution/ts/day16/tests/articleBuilder.ts
import {Article} from "../src/blog";
import {faker} from "@faker-js/faker";

export const Author = "Pablo Escobar";
export const CommentText = "Amazing article !!!";

export class ArticleBuilder {
    private comments: Record<string, string> = {};

    static anArticle(): ArticleBuilder {
        return new ArticleBuilder();
    }

    commented(): ArticleBuilder {
        this.comments[CommentText] = Author;
        return this;
    }

    build(): Article {
        let article = Article.create(faker.lorem.sentence(), faker.lorem.paragraph());
        Object.entries(this.comments).forEach(([text, author]) => article = article.addComment(text, author));
        return article;
    }
}